In [1]:
import sys
import os

# Add the root directory to sys.path
sys.path.append(os.path.abspath('..'))

from scripts import UserSatisfaction


# customer Satisfaction

In [2]:
satisfaction = UserSatisfaction()

In [3]:
satisfaction.calculate_engagemnt_score()

MSISDN/Number  engagement_score
0        3.360100e+10      7.169895e+08
1        3.360100e+10      1.539710e+09
2        3.360100e+10      8.836157e+08
3        3.360101e+10      1.529499e+09
4        3.360101e+10      2.455124e+08
...               ...               ...
106851   3.379000e+10      9.676300e+08
106852   3.379000e+10      1.288324e+09
106853   3.197021e+12      1.277697e+09
106854   3.370000e+14      1.237853e+09
106855   8.823971e+14      1.450607e+09

[106856 rows x 2 columns]

In [4]:
satisfaction.calculate_experiance_score()

MSISDN/Number  experience_score
0        3.360204e+10      4.903633e+06
1        3.360204e+10      4.903633e+06
2        3.360204e+10      4.903633e+06
3        3.360204e+10      4.903633e+06
4        3.360204e+10      1.079784e+07
...               ...               ...
106851   3.373228e+10      1.537423e+07
106852   3.373228e+10      4.903633e+06
106853   3.373228e+10      4.903633e+06
106854   3.373228e+10      4.903633e+06
106855   3.373228e+10      4.903633e+06

[106856 rows x 2 columns]

In [5]:
satisfaction.calculate_satisfaction_score()

MSISDN/Number  engagement_score  experience_score  satisfaction_score
6042    3.361489e+10      1.417964e+10      1.530993e+07        7.097476e+09
13131   3.362632e+10      1.379383e+10      1.546517e+07        6.904650e+09
12785   3.362578e+10      1.252637e+10      1.546297e+07        6.270917e+09
75968   3.367588e+10      1.243076e+10      1.337572e+07        6.222069e+09
36657   3.365973e+10      1.173566e+10      1.546307e+07        5.875562e+09
62633   3.366646e+10      9.472373e+09      1.534727e+07        4.743860e+09
30320   3.365836e+10      9.279878e+09      1.456718e+07        4.647223e+09
271     3.360313e+10      9.097869e+09      1.499078e+07        4.556430e+09
86060   3.369879e+10      9.049704e+09      1.378173e+07        4.531743e+09
13599   3.362708e+10      8.701151e+09      1.548078e+07        4.358316e+09

In [6]:
model = satisfaction.predict_satisfaction()
model

mean squared Error: 2.4123339955529603e-13


LinearRegression()

In [7]:
satisfaction.k_means_clustering_on_scores()

MSISDN/Number  satisfaction_cluster
0       3.360204e+10                     0
1       3.360204e+10                     0
2       3.360204e+10                     1
3       3.360205e+10                     0
4       3.360206e+10                     1
...              ...                   ...
89901   3.370000e+10                     1
89902   3.370000e+10                     1
89903   3.370000e+10                     0
89904   3.370000e+10                     0
89905   3.370000e+10                     0

[89906 rows x 2 columns]

In [8]:
satisfaction.aggergate_scores_per_cluster()

satisfaction_cluster engagement_score experience_score satisfaction_score
                                   mean             mean               mean
0                    0     1.326294e+09     7.770382e+06       6.670322e+08
1                    1     5.584975e+08     8.341035e+06       2.834192e+08

he final aggregated aggregate_scores_per_cluster DataFrame shows the mean values of engagement, experience, and satisfaction scores for each cluster. Here's an interpretation of the results:
Cluster 0:

    Engagement Score (Mean): 1.33×1091.33×109
    Experience Score (Mean): 7.77×1067.77×106
    Satisfaction Score (Mean): 6.67×1086.67×108

Cluster 1:

    Engagement Score (Mean): 5.58×1085.58×108
    Experience Score (Mean): 8.34×1068.34×106
    Satisfaction Score (Mean): 2.83×1082.83×108

Insights:

    Cluster 0 has a higher mean engagement score and satisfaction score compared to Cluster 1, indicating that users in Cluster 0 are more engaged and more satisfied overall.
    Experience Score is slightly lower in Cluster 0 compared to Cluster 1, suggesting that while users in Cluster 0 are more engaged, their overall experience might be slightly worse than those in Cluster 1.

Potential Actions:

    Focus on Cluster 1: Consider strategies to increase the engagement and satisfaction scores for users in Cluster 1, potentially improving overall user satisfaction.
    Experience Optimization for Cluster 0: Even though Cluster 0 users are more engaged, enhancing their experience could further boost their satisfaction.

These insights can guide further analysis and targeted interventions to improve user satisfaction.

sat

In [9]:
satisfaction.save_to_database()

Data saved successfully to table 'combined_scores'.
